store/store_download.go

```diff
@@ -646,16 +692,20 @@ var applyDelta = func(name string, deltaPath string, deltaInfo *snap.DeltaInfo,
 	partialTargetPath := targetPath + ".partial"
 
 	xdelta3Args := []string{"-d", "-s", snapPath, deltaPath, partialTargetPath}
-	cmd, err := getXdelta3Cmd(xdelta3Args...)
-	if err != nil {
-		return err
+
+	// sanity check that deltas are available and that the path for the xdelta3
+	// command is set
+	if ok := s.useDeltas(); !ok {
+		return fmt.Errorf("internal error: applyDelta used when deltas are not available")
 	}
 
-	if err := cmd.Run(); err != nil {
+	// run the xdelta3 command, cleaning up if we fail and logging about it
+	if runErr := exec.Command(s.xdelta3CmdLocation, xdelta3Args...).Run(); runErr != nil {
```

Comment:
This may be too simple for the case when we use the xdelta3 from the core/snapd snap. The CommandFromSystemSnap() is actually running it through `/lib64/ld-linux-x86-64.so.2` and adding a --library-path that will point to the core/snapd snap.